translator/translate/otel/common/eksdetector.go

```diff
@@ -0,0 +1,91 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type Detector interface {
+	getConfigMap(namespace string, name string) (map[string]string, error)
+}
+
+type EksDetector struct {
+	Clientset kubernetes.Interface
+}
+
+const (
+	authConfigNamespace = "kube-system"
+	authConfigConfigMap = "aws-auth"
+)
+
+var _ Detector = (*EksDetector)(nil)
+
+var (
+	detectorSingleton Detector
+	once              sync.Once
+)
+
+var (
+	getInClusterConfig  = func() (*rest.Config, error) { return rest.InClusterConfig() }
+	getKubernetesClient = func(confs *rest.Config) (kubernetes.Interface, error) { return kubernetes.NewForConfig(confs) }
+	// NewDetector creates a new singleton detector for EKS
+	NewDetector = func() (Detector, error) {
+		var errors error
+		once.Do(func() {
+			if clientset, err := getClient(); err != nil {
+				errors = err
+			} else {
+				detectorSingleton = &EksDetector{Clientset: clientset}
+			}
+		})
+
+		return detectorSingleton, errors
+	}
+)
+
+// IsEKS checks if the agent is running on EKS. This is done by using the kubernetes API to determine if the aws-auth
+// configmap exists in the kube-system namespace
+func IsEKS(eksDetector Detector) bool {
```

Comment:
Is it possible to have this just be `IsEKS()` so that we do not need to initialize a detector when it is just going to be a singleton anyways? 

I assume the first call of `IsEKS()` will initialize the singleton, and then future calls with re-use it

Reply:
Yup, was following the pattern of the resourcedetection processor, but this works better